src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -923,7 +954,11 @@ private static void ParseHeaderNameValue(Span<byte> line, HttpResponseMessage re
 
             // Note we ignore the return value from TryAddWithoutValidation; 
             // if the header can't be added, we silently drop it.
-            if (descriptor.HeaderType == HttpHeaderType.Content)
+            if (isFromTrailer)
+            {
+                response.TrailingHeaders.TryAddWithoutValidation(descriptor, headerValue);
```

Comment:
Do we need to do the same `descriptor.HeaderType == HttpHeaderType.Request ? descriptor.AsCustomHeader() : descriptor` dance that's done in the else block below?

Reply:
It is not clear to me since the headers go to their own collection. I made that change so it is consistent with "normal" headers.